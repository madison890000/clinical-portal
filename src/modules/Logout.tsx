import { Button } from '@mui/material';
import { useNavigate } from 'react-router';
import { useContext } from 'react';
import { AppContext } from '../contexts/AppContext';

const Logout = () => {
    const { logout } = useContext(AppContext);
    const navigate = useNavigate();
    const logoutHandle = () => {
        window.sessionStorage.clear();
        logout();
        navigate('/login');
    };
    return (
        <Button variant="outlined" onClick={logoutHandle}>
            Logout
        </Button>
    );
};
export default Logout;
