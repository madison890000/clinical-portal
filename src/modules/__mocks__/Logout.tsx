import { Button } from '@mui/material';

const MockLogout = () => {
    const logoutHandle = () => {};
    return (
        <Button variant="outlined" onClick={logoutHandle}>
            Logout
        </Button>
    );
};
export default MockLogout;
