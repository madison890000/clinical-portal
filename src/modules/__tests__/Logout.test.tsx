import React from 'react';
import Logout from '../Logout';
import { render, waitFor } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

jest.mock('../../services', () => {
    return {
        getPatientById: () => {
            return Promise.resolve({
                title: 'Mr.',
                firstName: 'James',
                preferredName: 'Jay',
                middleName: 'Shaun',
                familyName: 'Cooper',
                age: 17,
                sex: 'Indeterminate'
            });
        }
    };
});
test('renders Logout correctly', async () => {
    const { container } = render(
        <BrowserRouter>
            <Logout />
        </BrowserRouter>
    );
    await waitFor(() => {
        expect(container).toMatchSnapshot();
    });
});
