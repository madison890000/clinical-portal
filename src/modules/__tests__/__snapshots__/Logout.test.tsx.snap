// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Logout correctly 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Logout
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
